src/components/transactions/history/List.tsx

```diff
@@ -53,19 +49,29 @@ export function List({ setIsScrolled, targetRef }: Props) {
         onChange: () => hasNextPage && fetchNextPage(),
     });
 
-    const baseTx = useMemo(() => data?.pages.flatMap((p) => p) || [], [data?.pages]);
+    const baseTx = useMemo(() => {
+        // We are caching converted transactions to avoid re-converting all of them on every execution
+        // If someone have 200 transactions it is more efficient to convert only new ~20 then 200
+        const sanitizedData = data?.pages.flatMap((page) => page) || [];
+        const newTransactions = sanitizedData.slice(convertedTransactions.current.length);
+        const converted = newTransactions.map((transaction) =>
+            convertWalletTransactionToCombinedTransaction(transaction)
+        );
+        convertedTransactions.current = [...convertedTransactions.current, ...converted];
+        return convertedTransactions.current;
+    }, [data?.pages.length]); // Re-run only when the number of pages changes
```

Comment:
the comment helps and it makes sense but i hate leaving the warning in 😬 i don't have any bridge txs to test with myself but can't we split this out or something so we don't have a "missing" dep?